processor/lsmintervalprocessor/internal/merger/value.go

```diff
@@ -0,0 +1,796 @@
+					return m.ExponentialHistogram().DataPoints().Len() == 0
+				case pmetric.MetricTypeSummary:
+					return m.Summary().DataPoints().Len() == 0
+				}
+				return false
+			})
+			return sm.Metrics().Len() == 0
+		})
+		return rm.ScopeMetrics().Len() == 0
+	})
+	return s.source, nil
+}
+
+// merge merges the provided value to the current value instance.
+func (v *Value) merge(op Value) error {
+	for mOtherID, mOther := range op.metricLookup {
+		resOtherID := mOtherID.Resource()
+		scopeOtherID := mOtherID.Scope()
+		resOther := op.resLookup[resOtherID]
+		scopeOther := op.scopeLookup[scopeOtherID]
+
+		// Merge/add resource metrics. Note that if the resource metrics
+		// overflows then the ID will be different from the other resource ID.
+		resID, err := v.addResourceMetrics(resOther.ResourceMetrics)
+		if err != nil {
+			return fmt.Errorf("failed to merge resource metrics: %w", err)
+		}
+		// Merge/add scope metrics. Note that if the scope metrics overflows
+		// then the ID will be different from the other scope ID.
+		scopeID, err := v.addScopeMetrics(resID, scopeOther.ScopeMetrics)
+		if err != nil {
+			return fmt.Errorf("failed to merge scope metrics: %w", err)
+		}
+		// Merge any overflow estimators for scope or datapoints. Note that here
+		// we assume that the limits for both metrics being merged are identical
+		// and thus if any of the metric has overflowed then the target metric
+		// for merge will definitely overflow. Thus, merging the estimators is
+		// safe and required to correctly estimate the total number of overflow.
+		// Note that overflow merging is not deterministic and the estimates are
+		// always estimates as it is possible that the oveflows have hashes which
+		// haven't overflowed in the source metric.
+		res := v.resLookup[resID]
+		if err := res.scopeLimits.MergeEstimators(resOther.scopeLimits); err != nil {
+			return fmt.Errorf("failed to merge scope overflow estimators: %w", err)
+		}
+		scope := v.scopeLookup[scopeID]
+		if err := scope.datapointsLimits.MergeEstimators(scopeOther.datapointsLimits); err != nil {
+			return fmt.Errorf("failed to merge datapoints overflow estimators: %w", err)
+		}
+
+		// Finally merge the metric
+		v.mergeMetric(resID, scopeID, mOther)
+	}
+	// Merge any resource overflow estimators
+	if err := v.resourceLimits.MergeEstimators(op.resourceLimits); err != nil {
+		return fmt.Errorf("failed to merge resource overflow estimators: %w", err)
+	}
+	return nil
+}
+
+// addResourceMetrics adds a new resource metrics to the store while also
+// applying resource limiters. If a limit is configured and breached by
+// adding the provided resource metric, then, a new overflow resource
+// metric is created and returned.
+func (s *Value) addResourceMetrics(
+	otherRm pmetric.ResourceMetrics,
+) (identity.Resource, error) {
+	resID := identity.OfResource(otherRm.Resource())
+	if _, ok := s.resLookup[resID]; ok {
+		return resID, nil
```

Comment:
Is it possible that two ResourceMetrics have identical Resources, but different SchemaUrls? Do we need to consider this? Same for ScopeMetrics

Reply:
>  Do we need to consider this?

I don't think so, at least not for now I think since even the identity packages are not considering this.